feat(delete): Add delete method to postcards, letters, and checks

scotthasbrouck opened the pull request:
Adds `delete()` method to: `/letters`, `/postcards`, and `/checks` resources

Example useage:
```
Lob.postcards.delete('psc_xxxxxxxxxx', function (err, res) {
  if (err) { return console.log(err.message); }
  console.log('Postcard Deleted! ' + res.id);
 });
```

One integration test of the delete method on each `/letters`, `/postcards`, and `/checks` resource

robinjoseph08 commented on .gitignore:
```diff
@@ -15,3 +15,5 @@ node_modules
 
 coverage
+
+*.sublime-*
```
For all of our files in every since repo, we like to have a new line at the end of every line. Our linter usually catches that, but since this isn't a `.js` file, it's not run against it. [Here's a pretty good SO answer](http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) for why it's useful/common. It's unfortunate that Sublime's default is to not add the last `\n`. But you can add `"ensure_newline_at_eof_on_save": true` to your preferences to make sure that Sublime does add it if it doesn't exist. While you're at it, I would also recommend adding `"trim_trailing_white_space_on_save": true` as well since our linter checks for that as well.

scotthasbrouck replied:
Good catch. I normally have this i my sublime config, but I'm still getting setup on my new machine and havent copied over my preferences. I'll also have to catch up with you on the lob eslint config. When I run it I get about 100 linter errors that were there before I edited the code.

robinjoseph08 commented on lib/resources/checks.js:
```diff
@@ -21,6 +21,10 @@ util.inherits(Checks, ResourceBase);
 
+  this.delete = function (id, callback) {
+    return this._transmit('DELETE', id, null, null, callback);
+  }
```
Nit: there should be a semi colon at the end of this function expression (along with the `this.delete`s in the other files). The linter should've caught this... That makes me think that there's something amiss with how our linter is set up. I'll take a look to see what's up.

scotthasbrouck replied:
I actually forgot to run eslint before committing, but running it now see my above comment

robinjoseph08 commented on README.md:
```diff
@@ -153,18 +153,21 @@ Lob.addresses.retrieve('adr_bad_id')
     - [The Postcard Object](https://lob.com/docs/node#postcards_object)
     - [Create a Postcard](https://lob.com/docs/node#postcards_create)
     - [Retrieve a Postcard](https://lob.com/docs/node#postcards_retrieve)
+    - [Delete a Postcard](https://lob.com/docs/node#postcards_delete)
```
@ami is this the ordering that we want (object, create, retrieve, delete, list)? I feel like it should mimic how the docs page will be setup. Also, should we be linking to them just yet since these links don't lead to anywhere?

ami replied:
@scotthasbrouck let's leave the commit with the links out of this PR for now - we can go ahead and merge the rest. we can add the links later once the lob.com side is live.

leore commented:
@ami - I thought you wanted to do it in the wrappers first?

On Mon, Mar 20, 2017 at 9:45 AM, Ami Wang <notifications@github.com> wrote:

> *@ami* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/lob/lob-node/pull/175#discussion_r106952665>:
>
> > @@ -153,18 +153,21 @@ Lob.addresses.retrieve('adr_bad_id')
>      - [The Postcard Object](https://lob.com/docs/node#postcards_object)
>      - [Create a Postcard](https://lob.com/docs/node#postcards_create)
>      - [Retrieve a Postcard](https://lob.com/docs/node#postcards_retrieve)
> +    - [Delete a Postcard](https://lob.com/docs/node#postcards_delete)
>
> @scotthasbrouck <https://github.com/scotthasbrouck> let's leave the
> commit with the links out of this PR for now - we can go ahead and merge
> the rest. we can add the links later once the lob.com side is live.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/lob/lob-node/pull/175#discussion_r106952665>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABqWc_ddPrQ1ZFpEEAcLcP0AHCloZpkDks5rnq0ugaJpZM4MhK8h>
> .
>



-- 
Leore Avidar
CEO & Co-founder
(847) 630-9275
leore@lob.com

ami commented:
@leore - yes, what i mean above is JUST the links to the docs in the readme. no point putting in links that don't function yet, right?

leore commented:
For one day it doesn't matter and saves scott from having to do another PR. Lets not put any blockers even as small as this.

ami commented:
since there are already other changes that need to be made (see comments above), this can easily go with it. it's a bad experience to have broken links. plus, README changes don't need another PR, they can just be merged directly to master. either scott or myself can easily go ahead and do that across all wrappers (not just this one) once the lob.com docs side is released.

scotthasbrouck commented:
Hey all - thanks for all the feedback! I'll get to work on this and make a new commit to resolve everything. It's really no more than a minute to add those links back in. I'll just comment out the links so they'll be there waiting in the future when the docs are updated

mgartner commented on package.json:
```diff
@@ -10,7 +10,7 @@
     "printing"
   ],
-  "version": "3.9.0",
+  "version": "3.10.0",
```
You shouldn't update this version manually. Our release script will automatically update it for you.

mgartner commented on CHANGELOG.md:
```diff
@@ -1,3 +1,28 @@
+### 3.10.0 (2017-3-20)
```
The changelog should get updated outside of a PR, after all changes have be merged to master. And it can be run on its own without a review.

mgartner commented:
@scotthasbrouck can you address the comments above?

scotthasbrouck commented:
@mgartner fixed the change log and commit message. Tests are passing.

dmlittle commented:
@scotthasbrouck can you rebase this branch of master? There are a lot of files that are already on master (the addition of USVerification and IntlVerification endpoints are in this PR and those changes have already been merged into master). 

Also, the commit message should be about the addition of the delete endpoints.

scotthasbrouck commented:
@dmlittle the commit message did have the feature in it, but it was on line 2 beneath a chore. Just changed the order so the feature comment is on the top line.
Also now rebased of master

peterdolan commented:
@scotthasbrouck: Should this be assigned to donald/marcus?

scotthasbrouck commented:
@dmlittle cool thanks!
My vote to put it in there is it won't make a difference to anyone who doesn't use Sublime, and if we accept any PR's from someone who uses it then it'll keep the repo clean by default
